internal/inject/inject_test.go

```diff
@@ -0,0 +1,228 @@
+package inject
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/jarcoal/httpmock"
+	"github.com/luthermonson/go-proxmox"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/cloudinit"
+	"github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/ignition"
+	"github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/proxmox/goproxmox"
+)
+
+const (
+	testBaseURL   = "http://pve.local.test/" // regression test against trailing /
+	bootstrapData = `{
+  "ignition": {
+    "config": {},
+    "security": {
+      "tls": {}
+    },
+    "timeouts": {},
+    "version": "2.3.0"
+  },
+  "networkd": {},
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "sshAuthorizedKeys": [
+          "ssh-ed25519 ..."
+        ]
+      }
+    ]
+  },
+  "storage": {
+    "files": [
+      {
+        "filesystem": "root",
+        "path": "/etc/sudoers.d/core",
+        "contents": {
+          "source": "data:,core%20ALL%3D(ALL)%20NOPASSWD%3AALL%0A",
+          "verification": {}
+        },
+        "mode": 384
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "[Unit]\nDescription=kubeadm\n# Run only once. After successful run, this file is moved to /tmp/.\nConditionPathExists=/etc/kubeadm.yml\nAfter=network.target\n[Service]\n# To not restart the unit when it exits, as it is expected.\nType=oneshot\nExecStart=/etc/kubeadm.sh\n[Install]\nWantedBy=multi-user.target\n",
+        "enabled": true,
+        "name": "kubeadm.service"
+      }
+    ]
+  }
+}`
+)
+
+func TestISOInjectorInjectCloudInit(t *testing.T) {
+	client := newTestClient(t)
+
+	vm := &proxmox.VirtualMachine{
+		Node: "pve",
+		VMID: proxmox.StringOrUint64(100),
+		VirtualMachineConfig: &proxmox.VirtualMachineConfig{
+			Agent:     "1",
+			TagsSlice: []string{"flatcar"},
+			Tags:      "flatcar",
+		},
+	}
+
+	httpmock.RegisterResponder(http.MethodGet, fmt.Sprintf(`=~/nodes/%s/status`, "pve"),
+		newJSONResponder(200, proxmox.Node{Name: "pve"}, 2))
+
+	httpmock.RegisterResponder(http.MethodGet, fmt.Sprintf(`=~/nodes/%s/qemu/%d/status/current`, "pve", 100),
+		newJSONResponder(200, vm, 1))
+
+	httpmock.RegisterResponder(http.MethodGet, fmt.Sprintf(`=~/nodes/%s/qemu/%d/config`, "pve", 100),
+		newJSONResponder(200, vm.VirtualMachineConfig, 1))
+
+	vm, err := client.GetVM(context.Background(), "pve", 100)
+	require.NoError(t, err)
+
+	injector := &ISOInjector{
+		VirtualMachine: vm,
+		BootstrapData:  []byte(""),
+		MetaRenderer:   cloudinit.NewMetadata("xxx-xxxx", "my-custom-vm"),
+		NetworkRenderer: cloudinit.NewNetworkConfig([]cloudinit.NetworkConfigData{
+			{
+				Name:       "eth0",
+				IPAddress:  "10.1.1.6/24",
+				Gateway:    "10.1.1.1",
+				DNSServers: []string{"8.8.8.8", "8.8.4.4"},
+			}}),
```

Comment:
[optional][formatting] Should be wrapped for better formatting

Reply:
what do you mean? 
this is the default go formatting btw

Reply:
Aye, the braces wrapped to match the level they're opened in… …saw a mix of both variants in the code and my personal opinion would be to use the wrapped variant though functionality wise it doesn't matter.

Reply:
please check if the new formatting is okay?

Reply:
Perfect! 🙂